### DISCUSSION TITLE
After upgrading to 1.3.1, the Open Web UI always shows the "502: Bad Gateway" page.

### DISCUSSION THREAD

**Discussion by peterhow11** [llamafilm/tesla-http-proxy-addon#18] [General]
I may have missed something in the discussion but this now happens to me using iOS and Mac.  Running HAOS 2024.2.5. 

Here's the log after restarting:

      s6-rc: info: service s6rc-oneshot-runner: starting
      s6-rc: info: service s6rc-oneshot-runner successfully started
      s6-rc: info: service fix-attrs: starting
      s6-rc: info: service fix-attrs successfully started
      s6-rc: info: service legacy-cont-init: starting
      s6-rc: info: service legacy-cont-init successfully started
      s6-rc: info: service legacy-services: starting
      s6-rc: info: service legacy-services successfully started
      [07:22:50] INFO: Found existing keypair
      [07:22:50] INFO: Starting Tesla HTTP Proxy
      
      Do not listen on a network interface without adding client authentication. Unauthorized clients may
      be used to create excessive traffic from your IP address to Tesla's servers, which Tesla may respond
      to by rate limiting or blocking your connections.
      2024-03-08T07:22:50+08:00 [debug] Creating proxy
      2024-03-08T07:22:50+08:00 [info ] Listening on 0.0.0.0:443

Here's the nginx_tesla.conf file contents:

         server {
              server_name tesla.howletts.com.au;
      
              ssl_certificate /ssl/fullchain.pem;
              ssl_certificate_key /ssl/privkey.pem;
      
              # dhparams file
              ssl_dhparam /data/dhparams.pem;
      
              listen 443 ssl http2;
              add_header Strict-Transport-Security "max-age=31536000; includeSubDomains" always;
      
              resolver 127.0.0.11;
              set $target c03d64a7-tesla-http-proxy;
              
              # temporary Flask app for initial auth
              location / {
                  proxy_pass http://$target:8099;
              }
      
              # static public key for Tesla
              location /.well-known/appspecific/com.tesla.3p.public-key.pem {
                  root /share/tesla;
                  try_files /com.tesla.3p.public-key.pem =404;
              }
          }

Should I uninstall and reinstall?  If un-installing, should I clear out and config files (eg nginx_tesla.conf).

Thanks for your advice.

**Comment by llamafilm** [comment]
What problem are you seeing? That output all looks correct, like you’re finished.
If you need to rerun the auth flow because you lost the refresh token, flip that toggle on in the addon config page and then restart it.

**Reply by peterhow11** [reply]
![Screenshot 2024-03-08 at 12 33 18 pm](https://github.com/llamafilm/tesla-http-proxy-addon/assets/58583329/d7d22591-b4c1-47a0-a420-177bbf2dc901)

When I click on the Open Web UI in the Addon, I immediately see the above screen.  I can close the dialog, click the back-arrow and it goes back to the Addon's Info page.  I have restarted Home Assistant and the whole machine. The results are the same.

I expected to be presented with the page with the four setup buttons. 

Going into the Configuration page, I noticed that the "Regenerate Telsa Authentication" toggle was already on. Restarting the add-on with this in either setting makes no difference.  Is this the setting that you suggested I use?

**Reply by llamafilm** [reply]
That web UI page is only shown during initial setup. After that, the log you showed above is correct. If that regenerate setting isn’t working, I think you maybe need to update to 1.3.1.

**Reply by peterhow11** [reply]
I had already upgraded to 1.3.1 so something else is not right. What's the best to start over... Should I uninstall and reinstall? If un-installing, should I also delete any config files (eg nginx_tesla.conf) and anything else?

**Reply by baylanger** [reply]
If you uninstall the add-on, I'd expect the nginx_tesla.conf to be deleted.

```
docker exec -ti  addon_core_nginx_proxy ls -l /share/nginx_proxy/nginx_tesla.conf
-rw-r--r--    1 root     root           761 Mar  4 11:37 /share/nginx_proxy/nginx_tesla.conf
```

If it doesn't get deleted, there might be an issue. Regardless how often the add-on gets removed/installed, the content of the file isn't changing.

For sure, removing the add-on deletes everything inside the container (access/refresh token, keys, ...). Per [this comment](https://github.com/llamafilm/tesla-http-proxy-addon/issues/17#issuecomment-1983777643), v1.3.0 had a bug that left the "Regenerate Telsa Authentication" on. It is safe to reinstall but you could also just disable the Regenerate and leave it like that. Aside the Regen that was enable, is there something you think is broken?

**Comment by baylanger** [comment]
llamafilm please correct me if I'm wrong.

The "Open Web UI" works only when Flask is active...  I'm not sure if Flask uses the same port as the official Tesla proxy? If so, that might explains the 502 - if it's not using the same port, perhaps Flash could stay up and accept a connection to display add-on info (time of last refresh token, all systems ok, etc)  & stats (# of request from Tesla ?) .. then if it's using the same port, perhaps there's an easy way to tweak T proxy to properly answer the call from the "Open Web UI".

I was away when I noticed this and got the same 502 error but once home:

```
TESLA_AUTH_TOKEN=$(docker exec -ti addon_c03d64a7_tesla_http_proxy cat access_token)
curl --cacert /share/tesla/selfsigned.pem --header "Authorization: Bearer $TESLA_AUTH_TOKEN" https://c03d64a7-tesla-http-proxy/api/1/vehicles
{"error":"token expired (401)"
```
Service is up ... token expired is another issue :)

**Comment by llamafilm** [comment]
> The "Open Web UI" works only when Flask is active... 

Correct

> perhaps Flask could stay up and accept a connection to display add-on info

Possible, but I don’t like the security risk and there’s no need for this after you finish the initial auth. Tesla proxy requests are already shown in the log. I think I’ll disable that eventually after we get more users and prove it’s stable.

If you want more details on refresh tokens, enable debug logging on the Tesla component and teslajsonpy.  


> tweak T proxy to properly answer the call from the "Open Web UI".

Not possible
